src/main/java/org/snomed/snowstorm/core/data/services/ConceptService.java

```diff
@@ -734,6 +742,21 @@ private void joinComponentsToConcepts(PersistedComponents persistedComponents, M
 		}
 		return existingConceptsMap;
 	}
+	
+	private Map<String, Concept> getExistingSourceConceptsForSave(Collection<Concept> concepts, Commit commit) {
+		Map<String, Concept> existingSourceConceptsMap = new HashMap<>();
+		final List<String> conceptIds = concepts.stream().filter(concept -> concept.getConceptId() != null).map(Concept::getConceptId).collect(Collectors.toList());
+		if (!conceptIds.isEmpty()) {
+			BranchTimepoint branchTimePoint = new BranchTimepoint(commit.getSourceBranchPath(), BranchTimepoint.DATE_FORMAT.format(commit.getTimepoint()));
```

Comment:
It would be safer to use `commit.getBranch().getBase()` rather than timepoint because it's possible to rebase onto a point on the parent branch which is not the latest, for example when upgrading an extension.

Reply:
I think I tried that and ended up receiving the copy of the concept before it had been versioned and it had the end element set on it.   But I'm not sure which branch base I used there, so I'll give that a test and see if it still passes.